haskell/test/Test/Ligo/RegistryDAO.hs

```diff
@@ -33,6 +33,70 @@ import Ligo.Util
 import Test.Ligo.BaseDAO.Common
   (checkTokenBalance, dummyFA2Contract, makeProposalKey, sendXtz, totalSupplyFromLedger)
 
+-- | Helper type for unpack/pack
+data RegistryDaoProposalMetadata
+  = Normal_proposal NormalProposal
+  | Update_receivers_proposal UpdateReceiverParam
+  | Configuration_proposal ConfigProposal
+  | Transfer_proposal TransferProposal
+
+instance HasAnnotation RegistryDaoProposalMetadata where
+  annOptions = baseDaoAnnOptions
+
+data UpdateReceiverParam
+  = Add_receivers [Address]
+  | Remove_receivers [Address]
+
+instance HasAnnotation UpdateReceiverParam where
+  annOptions = baseDaoAnnOptions
+
+data NormalProposal = NormalProposal
+  { npAgoraPostId :: Natural
+  , npRegistryDiff :: [(MText, Maybe MText)]
+  }
+
+instance HasAnnotation NormalProposal where
+  annOptions = baseDaoAnnOptions
+
+data TransferProposal = TransferProposal
+  { tpAgoraPostId :: Natural
+  , tpTranfers :: [DAO.TransferType]
+  }
+
+instance HasAnnotation TransferProposal where
+  annOptions = baseDaoAnnOptions
+
+data ConfigProposal = ConfigProposal
+  { cpFrozenScaleValue :: Maybe Natural
+  , cpFrozenExtraValue :: Maybe Natural
+  , cpSlashScaleValue :: Maybe Natural
+  , cpSlashDivisionValue :: Maybe Natural
+  , cpMaxProposalSize :: Maybe Natural
+  }
+
+instance HasAnnotation ConfigProposal where
+  annOptions = baseDaoAnnOptions
+
+
+-- Note: Somehow setting `[@layout:comb]` in ligo types does not work.
+-- Have to use `ligoLayout` instead.
```

Comment:
Were you able to tell what was going wrong?

Are the types produced by LIGO not respecting the layout selection?

Reply:
I can't really tell what's going on. I make sure to make the ligo and haskell type looks exactly the same, added `[@layout:comb]` to all the related ligo types, but it does not work (it always result in fail to unpack). 

I do check the result type from ligo, it looks fine for me. It is quite trick but if you want me to investigate more into this, I could try.

Reply:
No that's ok, was just wondering.

Using the `ligoLayout` is perfectly fine, we'll get back to this later if we need to.